core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -103,10 +104,13 @@ class NewHadoopRDD[K, V](
   private val confBroadcast = sc.broadcast(new SerializableConfiguration(_conf))
   // private val serializableConf = new SerializableWritable(_conf)
 
-  private val jobTrackerId: String = {
-    val formatter = new SimpleDateFormat("yyyyMMddHHmmss", Locale.US)
-    formatter.format(new Date())
-  }
+  @transient
+  private val dateTimeFormatter =
+    DateTimeFormatter
+      .ofPattern("yyyyMMddHHmmss", Locale.US)
+      .withZone(ZoneId.systemDefault())
+
+  private val jobTrackerId: String = dateTimeFormatter.format(new Date().toInstant)
```

Comment:
We dont need this field

```suggestion
  private val jobTrackerId: String = {
  val dateTimeFormatter =
      DateTimeFormatter
        .ofPattern("yyyyMMddHHmmss", Locale.US)
        .withZone(ZoneId.systemDefault())  
    dateTimeFormatter.format(new Date().toInstant)
  }
```